captum/optim/_core/loss.py

```diff
@@ -128,8 +128,16 @@ def loss_fn(module: ModuleOutputMapping) -> torch.Tensor:
 
         name = f"Compose({', '.join([self.__name__, other.__name__])})"
         target = (
-            self.target if hasattr(self.target, "__iter__") else [self.target]
-        ) + (other.target if hasattr(other.target, "__iter__") else [other.target])
+            self.target
+            if hasattr(self.target, "__iter__")
+            and not isinstance(self.target, nn.Module)
+            else [self.target]
+        ) + (
+            other.target
+            if hasattr(other.target, "__iter__")
+            and not isinstance(other.target, nn.Module)

```

Comment:
This of course will work! But we have already clearly defined the type of `self.target` as l`Union[nn.Module, List[nn.Module]]`. So we only need to check if it is a `list` and no need to worry about other `iterable`. We can just do
```py
if isinstance(self.target, list)
```

Reply:
Yeah, I think that the `hasattr(obj, __iter__)` is a holdover from when SK was putting together the first iteration of the comprehension system, when weren't 100% sure how it was going to work.

I've replaced the relevant lines with `isinstance(self.target, list)`!